### PR TITLE
updated modelbpa to use TOM

### PR DESCRIPTION
**Pull request by m-kovalsky**

### CODE REVIEW

File: src/sempy_labs/tom/_model.py [hunk 1]
```diff
@@ -1119,8 +1119,12 @@ def get_annotation_value(self, object, name: str):
         str
             The annotation value.
         """
+        try:
+            value = object.Annotations[name].Value
```
Comment (eisber) [inline src/sempy_labs/tom/_model.py:1123]: can't you check if name is object.annotations? catch generic exception masks real issues

File: src/sempy_labs/tom/_model.py [hunk 2]
```diff
@@ -1119,8 +1119,12 @@ def get_annotation_value(self, object, name: str):
...
+        except Exception:
+            value = ""
```
Comment (eisber) [inline src/sempy_labs/tom/_model.py:1125]: why not return None if the value really isn't there. can the value of the annotation by an empty string and/or None?

File: src/sempy_labs/tom/_model.py [hunk 3]
```diff
@@ -2814,6 +2825,9 @@ def records_per_segment(self, object: "TOM.Partition"):
                 object=object, name="Vertipaq_RecordsPerSegment"
             )
 
+        if result == "":
```
Comment (eisber) [inline src/sempy_labs/tom/_model.py:2828]: I'm wondering if you should follow a similar pattern to dict.get(key, default_value). 
if the annotation or extended property isn't there, return the default value.

**Review by eisber — CHANGES_REQUESTED** [review]

File: src/sempy_labs/tom/_model.py (hunk 2 above)
Comment (m-kovalsky) [reply src/sempy_labs/tom/_model.py:1125]: i made the else clause return None

File: src/sempy_labs/tom/_model.py (hunk 3 above)
Comment (m-kovalsky) [reply src/sempy_labs/tom/_model.py:2828]: i updated the functions to return the following:

```python
return int(result) if result is not None else 0
```

**Review by eisber — APPROVED** [review]

**Pull request merged**